front/src/modules/timesStops/TimesStops.tsx

```diff
@@ -0,0 +1,196 @@
+    isMarginValid: op.theoreticalMargin ? marginRegExValidation.test(op.theoreticalMargin) : true,
+    onStopSignal: op.onStopSignal || false,
+  }));
+  return rowOPs;
+};
+
+const deleteViaButton = (removeVia: () => void, isRowVia: boolean) => {
+  if (isRowVia) {
+    return (
+      <button type="button" onClick={removeVia}>
+        ❌
+      </button>
+    );
+  }
+  return <div />;
+};
+
+const TimesStops = ({ pathProperties, pathSteps = [], setPathProperties }: TimesStopsProps) => {
+  const { t } = useTranslation('timesStops');
+  const dispatch = useAppDispatch();
+  const { upsertViaFromSuggestedOP, updatePathSteps } = useOsrdConfActions();
+
+  const [timesStopsSteps, setTimesStopsSteps] = useState<PathWaypointColumn[]>([]);
+
+  useEffect(() => {
+    const suggestedOPs = formatSuggestOPsToRowOPs(pathProperties.allVias, t);
+    setTimesStopsSteps(suggestedOPs);
+  }, [pathSteps]);
+
+  const columns: Column<PathWaypointColumn>[] = useMemo(
+    () => [
+      {
+        ...keyColumn<PathWaypointColumn, 'name'>('name', createTextColumn()),
+        title: t('name'),
+        disabled: true,
+      },
+      {
+        ...keyColumn<PathWaypointColumn, 'ch'>('ch', createTextColumn()),
+        title: 'Ch',
+        disabled: true,
+        grow: 0.1,
+      },
+      {
+        ...keyColumn<PathWaypointColumn, 'arrival'>('arrival', timeColumn),
+        title: t('arrivalTime'),
+
+        // We should not be edit the arrival time of the origin
+        cellClassName: ({ rowIndex }) => (rowIndex === 0 ? 'dsg-hidden-cell' : ''),
+        grow: 0.6,
+      },
+      {
+        ...keyColumn<PathWaypointColumn, 'stopFor'>(
+          'stopFor',
+          createTextColumn({
+            continuousUpdates: false,
+            alignRight: true,
+          })
+        ),
+        title: `${t('stopTime')}`,
+        grow: 0.6,
+      },
+      {
+        ...keyColumn<PathWaypointColumn, 'onStopSignal'>(
+          'onStopSignal',
+          checkboxColumn as Partial<Column<boolean | undefined>>
+        ),
+        title: t('receptionOnClosedSignal'),
+        grow: 0.6,
+      },
+      {
+        ...keyColumn<PathWaypointColumn, 'theoreticalMargin'>(
+          'theoreticalMargin',
+          createTextColumn({
+            continuousUpdates: false,
+            alignRight: true,
+            placeholder: t('theoreticalMarginPlaceholder'),
+            formatBlurredInput: (value) => {
+              if (!value || value === 'none') return '';
+              if (!marginRegExValidation.test(value)) {
+                return `${value}${t('theoreticalMarginPlaceholder')}`;
+              }
+              return value;
+            },
+          })
+        ),
+        cellClassName: ({ rowData }) => (rowData.isMarginValid ? '' : 'invalidCell'),
+        title: t('theoreticalMargin'),
+        disabled: ({ rowIndex }) => rowIndex === findLastIndex(pathProperties.allVias),
+      },
+    ],
+    [t, pathProperties, timesStopsSteps]
```

Comment:
I think you can change these lines like this, this way you can update the dependencies of the useMemo to avoid excessive computations
```suggestion
        disabled: ({ rowIndex }) => rowIndex === pathProperties.allVias.length - 1,
      },
    ],
    [t, pathProperties.allVias.length]
```

Reply:
Done